token_block.block.inc

```diff
@@ -18,7 +18,14 @@ class TokenBlock extends Block {
    * {@inheritdoc}
    */
   function getTitle() {
-    return !empty($this->settings['title']) ? check_plain($this->settings['title']) : t('Token block');
+    if (!empty($this->settings['title'])) {
+      $entity_type = $this->childDelta;
+      $entity = $this->contexts[$entity_type]->data;
+      $data = array($entity_type => $entity);
+      return check_plain(token_replace($this->settings['title'], $data));
```

Comment:
The `check_plain` is not needed here. By default, `token_replace` will sanitize the token, unless otherwise specified. That sanitization happens in the hook_tokens implementations. For example, see `node_tokens()`.

Reply:
I added the check_plain because the token_replace included html <p> tags. In my use case the html ended up causing an issue when the token block was set as a layout title. When I saw that issue, my immediate thought was that block titles should always be plain text with no html, but I didn't think it through from a general perspective. 

Is there a case that html should be included in a block title?